roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -25,24 +26,27 @@
 #include <string>
 
 /// @brief  A wrapper class to store a C++ function of type 'double (*)(double* )'.
-/// The parameters can be accessed as x[<relative position of param in paramSet>] in the function body.
+/// The parameters can be accessed as params[<relative position of param in paramSet>] in the function body.
+/// The observables can be accessed as obs[i * num_entries + j], where i represents the observable position and j
+/// represents the data entry.
 /// @tparam Func Function pointer to the generated function.
-template <typename Func = double (*)(double *), typename Grad = void (*)(double *, double *)>
+template <typename Func = double (*)(double *, double *), typename Grad = void (*)(double *, double *, double *)>
```

Comment:
We have carried the `RooFuncWrapper` around for some time now, and at no point it became apparent why it would be useful to have these types templated.

Can we not just define these types in the class, like `using Func = double (*)(double *, double *)` (in `private`) and then have the class split up in a `.h` and `.cxx` file?

If you follow up with doing that, please do it in a separate commit so it's not mixed with the other changes.

Reply:
I remember having a conversation where you suggested we should keep the templated types around for now :) But sure we could move them too.

Reply:
At that time I didn't understand yet that the signature will always be the same, sorry! I thought it would still depend on the number of parameters and observables,

Reply:
yes, I agree! It was not as clear what the types would be then. Once you are done with the review, I can add another commit that does this refactoring.